test/e2e/network_test.go

```diff
@@ -368,21 +368,24 @@ func getAffinityForNode(node *v1.Node) (*v1.Affinity, error) {
 	}, nil
 }
 
-// ensureNamespace checks if a namespace with the provided name exists and creates one if it does not
-func (tc *testContext) ensureNamespace(name string) error {
+// ensureNamespace checks if a namespace with the provided name exists and creates one if it does not with the given
+// labels
+func (tc *testContext) ensureNamespace(name string, labels map[string]string) error {
 	// Check if the namespace exists
 	_, err := tc.client.K8s.CoreV1().Namespaces().Get(context.TODO(), name, metav1.GetOptions{})
 	if err != nil && !apierrors.IsNotFound(err) {
 		return err
 	}
 	if err == nil {
+		// TODO: ensure required labels
```

Comment:
Will this be a valid corner case? What is the probability that another entity created the `wmco-test` namespace?

Reply:
Makes sense.